cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ConsoleSettingsHttpHandler.java

```diff
@@ -61,7 +62,11 @@ public ConsoleSettingsHttpHandler(Authenticator authenticator, ConsoleSettingsSt
   @Path("/")
   @GET
   public void get(HttpRequest request, HttpResponder responder) throws Exception {
-    String userId = getAuthenticatedAccountId(request);
+    String userId = SecurityRequestContext.getUserId();
```

Comment:
I'd suggest changing `SecurityRequestContext.getUserId()` to return an `Optional`, and then it's clear when `userId` is missing. Then you can retrieve the userId as follows:

```
String userId = SecurityRequestContext.getUserId().or("");
```